index.bs

```diff
@@ -8972,6 +9018,373 @@ worker=] algorithm:
+organize persistent data such as [=cookies=] and local storage.
+
+A <dfn>storage partition key</dfn> uniquely identifies a [=storage partition=].
+
+### Definition ### {#module-storage-definition}
+
+[=Remote end definition=]
+
+<pre class="cddl remote-cddl">
+StorageCommand = (
+  storage.GetCookies //
+  storage.SetCookie //
+  storage.DeleteCookies
+)
+</pre>
+
+[=Local end definition=]
+
+<pre class="cddl local-cddl">
+StorageResult = (
+  storage.GetCookiesResult //
+  storage.SetCookieResult //
+  storage.DeleteCookiesResult
+)
+</pre>
+
+### Types ### {#module-storage-types}
+
+#### The storage.PartitionKey Type #### {#type-storage-PartitionKey}
+
+[=Remote end definition=] and [=Local end definition=]
+
+<pre class="cddl local-cddl remote-cddl">
+storage.PartitionKey = {
+    ? userContext: text,
+    ? sourceOrigin: text,
+    Extensible,
+}
+</pre>
+
+The <code>storage.PartitionKey</code> type represents a [=storage partition key=].
+
+A [=remote end=] has a [=/set=] of <dfn>storage partition key attributes</dfn> which contains zero or more implementation-defined string values.
+
+A [=remote end=] has a [=/map=] of <dfn>default values for storage partition key attributes</dfn> which contains zero or more implementation-defined entries. Each key must be a member of the [=storage partition key attributes=] which the [=remote end=] does not require as input to [=storage.getCookies=], [=storage.setCookie=], and [=storage.deleteCookies=]; the corresponding value is a the value which the [=remote end=] uses when one is not provided.
+
+<div algorithm>
+To <dfn>expand a storage partition spec</dfn> given |partition spec| and |cookie domain|:
+
+1. If |partition spec| is a [=browsing context id=]:
+
+  1. Let |browsing context| be the result of [=trying=] to [=get a browsing context=] given |partition spec|.
+
+  1. Let |partition key| be the [=storage partition key|key=] of |browsing context|'s [=associated storage partition=].
+
+  1. Return [=success=] with data |partition key|.
+
+1. Otherwise, if |partition spec| is null:
+
+  1. Let |partition key| be an empty [=/map=].
+
+1. Otherwise:
+
+  1. Assert: |partition spec| is a [=/map=] with zero or more entries.
+
+  1. Let |partition key| be |partition spec|.
+
+1. For each |name| → |default value| in the [=default values for storage partition key attributes=]:
+
+  1. If |partition key|[|name|] does not [=map/exist=]:
+
+    1. [=map/Set=] |partition key|[|name|] to |default value|.
+
+1. If |partition key|["<code>sourceOrigin</code>"] does not [=map/exist=]:
+
+  1. If |cookie domain| is null, then:
+
+    1. Return [=error=] with [=error code=] [=unknown source origin=].
+
+  1. [=map/Set=] |partition key|["<code>sourceOrigin</code>"] to |cookie domain|.
+
+1. Return [=success=] with data |partition key|.
+
+</div>
+
+<div algorithm>
+To <dfn>get the cookie store</dfn> given |storage partition key|:
+
+1. If a [=cookie store=] |match| exists for |storage partition key|:
```

Comment:
Here, I intended "match" as a noun (not a verb), but I didn't recognize the confusion that could result from using a homonym here. This operation is right on the edge of what we're normatively specifying, so I think this is best resolved with more detail rather than a simple word swap. [I've made an attempt along those lines](https://github.com/w3c/webdriver-bidi/pull/501/commits/d7e61f03357e1852259b4521d2b8e56f0d1ffba9), please let me know what you think!